### PR TITLE
feat(rust-sdk):add sync public api

### PR DESCRIPTION
**Pull request by thomsoe84**

### CODE REVIEW

**Review by adityasuseno — APPROVED** [review]
Looks Good to Me  ✅️

File: rust/src/main.rs [hunk 1]
```diff
@@ -1,10 +1,53 @@
...
+mod public;
 
-fn main() {
-    println!("{}", config::get_public_api_endpoint(&"SERVERTIME"));
+use std::error::Error;
+//use public::config;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    /*
+    println!("{:?}", config::get_public_api_endpoint(&"SERVERTIME"));
```
Comment (dendisuhubdy) [inline rust/src/main.rs:8]: rmove comments

File: rust/src/public/config.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-pub fn timeout() -> u8 {
+pub fn timeout() -> u64 {
```
Comment (dendisuhubdy) [inline rust/src/public/config.rs:1]: why u64?

File: rust/src/public.rs [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    match res.status() {
+        reqwest::StatusCode::OK => {
+            println!("Status: {} -> {}", res.status(), "Successfully retrieved announcements");
+            println!("{}", body);
+        },
+        _ => {
+            panic!("{} -> {}", res.status(), "Error, please check again your request");
+        },
+    };
+
```
Comment (dendisuhubdy) [inline rust/src/public.rs:196]: harusnya ada patternnya nih @thomsoe84 https://en.wikipedia.org/wiki/Software_design_pattern

banyak repeated code

**Review by dendisuhubdy — CHANGES_REQUESTED** [review]

**Pull request closed without merging**